pkg/apis/agones/v1/gameserver.go

```diff
@@ -461,25 +495,44 @@ func (gss *GameServerSpec) Validate(devAddress string) ([]metav1.StatusCause, bo
 				}
 			}
 		}
-		for _, c := range gss.Template.Spec.Containers {
-			validationErrors := validateResources(c)
-			for _, err := range validationErrors {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Field:   "container",
-					Message: err.Error(),
-				})
-			}
-		}
 	}
-	objMetaCauses := validateObjectMeta(&gss.Template.ObjectMeta)
```

Comment:
The validation of the ObjectMeta field is handled by `k8s_validation.ValidatePodTemplateSpec` method.